### DISCUSSION TITLE
Examples of using Conduits and Conduit Keys

### DISCUSSION THREAD

**Discussion by jdaly13** [ProjectOpenSea/seaport-js#400] [Q&A]
Hello, thank you for all the work you've done, I had a question regarding the usage and implementation of Conduits and Conduit keys.  I was wondering if there were any example first off secondly I had some other questions.

First off I want to make sure I understand the methods and meanings clearly.  A Conduit would be another contract that implements the final transfers of Offers?  What other role can a conduit have?

It seems like to implement a Conduit via seaport JS -  When I Instantiate new Seaport in the second Param which is an object I would provide a conduitKeyToConduit object which there is a conduitkey and conduitValue which would be a contract
so it would look something like this

```
      const seaportWithSigner = new Seaport(signer, {
        conduitKeyToConduit: {
          "random": {myContractAddress}
        },
      });
```

but you can also provide a defaultConduitKey in the overrides Object so you would use both for example if you didn't want to inject into every createOrder? so it would look more like 
```
      const seaportWithSigner = new Seaport(signer, {
        conduitKeyToConduit: {
          "random":  {myContractAddress}
        },
        overrides: {
          defaultConduitKey: "random"
        },
      });
```

I'm going through all tests and every thing is documented well and code seems clear but I'm still having trouble wrapping my head around how to implement a conduit contract within seaport - if you have any examples or other documentation it would be helpful

**Comment by ryanio** [comment]
A conduit is where approvals are granted so it can be reused when new Seaport versions are deployed.

There is some info about the Conduit Controller here: https://docs.opensea.io/reference/seaport-conduit-controller

When a Seaport contract is deployed, it is deployed with only one allowed conduit controller. To create a new conduit, call createConduit on that conduit controller.

**Reply by jdaly13** [reply]
Thanks for your reply - I'm assuming you call createConduit via new Contract on ethers.js 
this address is the same no matter what chain for the contract address - 0x00000000F9490004C11Cef243f5400493c00Ad63
Where can I find the ABI? - When I create a contract instance In ethers I then call that method createConduit  and then I would take that info and use it when I instantiate seaportJS object?

Starting to understand - if you know of any tutorials/code examples - it would be helpful

> Note that the first twenty bytes of the supplied conduit key must match the caller and that a new conduit cannot be created if one has already been deployed using the same conduit key.

Is there an example of how to do this in JS?

**Reply by ryanio** [reply]
Yes the ConduitController is only at one address, you can find the ABI on etherscan toward the bottom of the page: https://etherscan.io/address/0x00000000F9490004C11Cef243f5400493c00Ad63#code You can also use the "Write Contract" tab to call createConduit

> When I create a contract instance In ethers I then call that method createConduit and then I would take that info and use it when I instantiate seaportJS object?

yes you would provide the conduit key

> Is there an example of how to do this in JS?

not really, it is not a common operation so there aren't helper methods in the sdk. you don't really need to create your own conduit, is there a reason you wouldn't use the OpenSea one? then your users won't need to redo approvals. also you can only submit orders to OpenSea using the OS conduit key

**Reply by jdaly13** [reply]
thanks @ryanio - I'm looking to use the conduit controller because I'm' trying to develop a proof of concept marketplace that could utilize and escrow contract and I feel using a conduit would give us more control over that - for example if an order is over a certain amount - funds could be held in escrow
